services/spar/test-integration/Test/Spar/DataSpec.hs

```diff
@@ -221,37 +225,39 @@ spec = do
           idp2' <- runSpar $ liftSem (IdPEffect.getConfig (idp1 ^. idpId))
           liftIO $ idp2' `shouldBe` Nothing
 
+-- TODO(sandy): This function should be more polymorphic over it's polysemy
+-- constraints than using 'RealInterpretation' in full anger.
```

Comment:
not sure that would be an improvement?  it's a local function, and the type is obviously flexible enough for it's use cases.

Reply:
The point being that it doesn't need all of `RealInterpreter`, it really just needs a way to run its constraints in `IO`. Making this more polymorphic would mean we could use the same function to test pure interpreters too.